### ISSUE TITLE
fix: replace runtime vmap_diag branch in JAX SVD backward with diag_embed

### ISSUE THREAD

**Issue by hunter-heidenreich** [hunter-heidenreich/Kabsch-Cookbook#31]
## Goal

Remove the Python-level conditional in the JAX SVD VJP backward that selects between `jnp.diag` and `jax.vmap(jnp.diag)` based on tensor rank at trace time.

## Background

In `jax/kabsch_svd_nd.py`, the VJP backward contains:

```python
# jax/kabsch_svd_nd.py:74-76
vmap_diag = jax.vmap(jnp.diag) if S.ndim > 1 else jnp.diag
S_diag = vmap_diag(grad_S)
S_mat = vmap_diag(S)
```

This is a Python conditional on `S.ndim`, which is evaluated at JAX trace time (inside `jax.jit`). The two branches produce structurally different JAX functions, meaning a JIT-compiled function compiled for batched input cannot be reused for unbatched input (and vice versa) -- they trace to different graphs. This is generally harmless (JAX retraces for new shapes) but it's unnecessary complexity.

The `jax.vmap(jnp.diag)` pattern is exactly what `jnp.diag_embed` is for -- it generalizes `diag` to batched inputs:

```python
# Replacement
S_diag = jnp.diag_embed(grad_S)  # works for any leading batch dims
S_mat = jnp.diag_embed(S)
```

(`jnp.diag_embed` requires JAX >= 0.4.1, which is already a transitive dependency.)

## Fix

Replace both occurrences in `_bwd`:

```python
# Before
vmap_diag = jax.vmap(jnp.diag) if S.ndim > 1 else jnp.diag
S_diag = vmap_diag(grad_S)
S_mat = vmap_diag(S)

# After
S_diag = jnp.diag_embed(grad_S)
S_mat = jnp.diag_embed(S)
```

## Acceptance Criteria

- `vmap_diag` conditional removed
- All existing JAX tests pass
- Gradient correctness tests pass (the diag is in the hot path of the backward)
- No JAX version compatibility regression (verify `jnp.diag_embed` availability)
